pkg/crosscluster/logical/txnmode/txnmode_test: TestTxnModeSmoketest failed

pkg/crosscluster/logical/txnmode/txnmode_test.TestTxnModeSmoketest [failed](https://mesolite.cluster.engflow.com/invocations/default/e9a60136-87db-4022-92bb-a53e885bc230?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL2xvZ2ljYWwvdHhubW9kZTp0eG5tb2RlX3Rlc3Q=) on master @ [596ed4c7e3a48146619358799898ac4a2face0e1](https://github.com/cockroachdb/cockroach/commits/596ed4c7e3a48146619358799898ac4a2face0e1):

Fatal error:

```
panic: pebble: closed [recovered, repanicked]
	panic: found 1 short-living non-stopped monitors in exec
	exec 20 KiB
	    flow-3064688e 20 KiB
	 [recovered]
	panic: found 1 short-living non-stopped monitors in exec
	exec 20 KiB
	    flow-3064688e 20 KiB
	

```
Stack: 

```
goroutine 8116 [running]:
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).doStop(0xc0045b6bb0, {0xd161690, 0xc00e97b2c0}, 0x0)
	pkg/util/mon/bytes_usage.go:781 +0xa9c
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).EmergencyStop(...)
	pkg/util/mon/bytes_usage.go:717
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).close(0xc00511e508, {0xd161690, 0xc00e97b2c0}, 0x1)
	pkg/sql/conn_executor.go:1520 +0xdf8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc00511e508, {0xd161690, 0xc00e97b2c0}, {0xb1e6400, 0xc014d7ad10})
	pkg/sql/conn_executor.go:1408 +0x227
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1({0xd161690, 0xc00e97b2c0}, {0x4b60a5?})
	pkg/sql/conn_executor.go:1126 +0x4b
panic({0xb1e6400?, 0xc014d7ad10?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).doStop(0xc0045b6bb0, {0xd161690, 0xc00fbcdb80}, 0x1)
	pkg/util/mon/bytes_usage.go:781 +0xa9c
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).Stop(...)
	pkg/util/mon/bytes_usage.go:722
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine.func1()
	pkg/sql/conn_executor_exec.go:2932 +0x3c5
panic({0xb2dbfe0?, 0xc0038bca20?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x81a
panic({0xb2dbfe0?, 0xc0038bca20?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/pebble.(*DB).applyInternal(0xc0043f4c08, 0xc02c047800, 0x133f6bbe, 0x0)
	external/com_github_cockroachdb_pebble/db.go:789 +0x9fd
github.com/cockroachdb/pebble.(*DB).Apply(...)
	external/com_github_cockroachdb_pebble/db.go:764
github.com/cockroachdb/pebble.(*DB).DeleteRange(0xc0043f4c08, {0xc01900d4d0, 0x1, 0x18}, {0xc0186174b8, 0x1, 0x1}, 0x133f6bbe)
	external/com_github_cockroachdb_pebble/db.go:667 +0xc5
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleMap).Clear(0xc001b634a0)
	pkg/storage/disk_map.go:170 +0x1b3
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleMap).Close(0xc001b634a0, {0xd161690, 0xc00fc04000})
	pkg/storage/disk_map.go:186 +0x45
github.com/cockroachdb/cockroach/pkg/crosscluster/producer.(*OrderedBuffer).Close(...)
	pkg/crosscluster/producer/ordered_buffer.go:278
github.com/cockroachdb/cockroach/pkg/crosscluster/producer.(*OrderedStreamHandler).Close(0xc0019154d0, {0xd161690, 0xc00fc04000})
	pkg/crosscluster/producer/ordered_event_stream.go:195 +0x10c
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*projectSetProcessor).close(0xc013faf888)
	pkg/sql/rowexec/project_set.go:356 +0x13c
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*projectSetProcessor).ConsumerClosed(0xc013faf888)
	pkg/sql/rowexec/project_set.go:365 +0x27
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).InternalClose(0xc0049edc08)
	pkg/sql/execinfra/processorsbase.go:983 +0x107
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Close(0xc0049edc08, {0xc00118f96f?, 0xc018617760?})
	pkg/sql/colexec/columnarizer.go:301 +0xb8
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*invariantsChecker).Close(0xc010a49630, {0xd160920, 0xc010a49540})
	pkg/sql/colexec/invariants_checker.go:137 +0x75
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecargs.(*CloserRegistry).Close.func1()
	pkg/sql/colexec/colexecargs/closer_registry.go:47 +0xd6
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc018617950)
	pkg/sql/colexecerror/error.go:162 +0x79
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecargs.(*CloserRegistry).Close(0xc002a10e40, {0xd160920, 0xc010a49540})
	pkg/sql/colexec/colexecargs/closer_registry.go:45 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).cleanup(0xc00f60e580, {0xd160920, 0xc010a49540})
	pkg/sql/colflow/vectorized_flow.go:674 +0x58
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Cleanup(0xc010e33e50, {0xd160920, 0xc010a49540})
	pkg/sql/colflow/vectorized_flow.go:386 +0x1f7
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run.func3()
	pkg/sql/distsql_running.go:1022 +0x59
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc00f34a6e0, {0xd160920, 0xc010a49540}, 0xc010d2a750, 0xc010d2a0d0, 0xc0022c6580, 0xc00eb7db08, 0xc00511f320, 0xc00ca5c698)
	pkg/sql/distsql_running.go:1088 +0x1c53
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc00f34a6e0, {0xd161690, 0xc00fbcdb80}, 0xc00511f320, 0xc010d2a750, 0xc010d2a0d0, {{0xd1a8d90, 0xc012ec6480}, 0x0}, 0xc00eb7db08, ...)
	pkg/sql/distsql_running.go:2397 +0x3a5
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll.func3(0xc00511ed48, 0xc00f34a6e0, {0xd161690, 0xc00fbcdb80}, 0xc00511f320, 0xc010d2a750, 0xc00eb7db08)
	pkg/sql/distsql_running.go:2096 +0x19d
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll(0xc00f34a6e0, {0xd161690, 0xc00fbcdb80}, 0xc00511f320, 0xc010d2a750, 0xc00511ed48, 0xc00eb7db08, 0x0)
	pkg/sql/distsql_running.go:2099 +0x359
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc00511e508, {0xd161690, 0xc00fbcdb80}, 0xc00511ed48, 0x3, {0x7f87054eb9a0, 0xc0037fcb60}, 0x0, 0xc004c175d8, 0x0)
	pkg/sql/conn_executor_exec.go:3694 +0xba5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00511e508, {0xd161690, 0xc00fbcdb80}, 0xc00511ed48, {0x7f87054eb9a0, 0xc0037fcb60})
	pkg/sql/conn_executor_exec.go:3185 +0x270e
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00511e508, {0xd160920, 0xc010a490e0}, {{0xd18ff40, 0xc0105c0e10}, {0x0, 0x0, 0x0}, {0xc002c5e073, 0x34}, ...}, ...)
	pkg/sql/conn_executor_exec.go:1178 +0x69b6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func2({0xd160920, 0xc010a490e0})
	pkg/sql/conn_executor_exec.go:183 +0x185
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0xc00511e508, {0xd160920, 0xc010a490e0}, {0xd18ff40, 0xc0105c0e10}, 0xc01152a200, 0xc00ca5eb20)
	pkg/sql/conn_executor_exec.go:4766 +0x997
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00511e508, {0xd160920, 0xc010a490e0}, {{0xd18ff40, 0xc0105c0e10}, {0x0, 0x0, 0x0}, {0xc002c5e073, 0x34}, ...}, ...)
	pkg/sql/conn_executor_exec.go:182 +0xdbe
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execPortal(0xc00511e508, {0xd160920, 0xc010a490e0}, {{0xc002c5e0e6, 0x0}, 0xc01152a200, {0xc0137baea0, 0x2, 0x2}, {0xc00dbf97ee, ...}, ...}, ...)
	pkg/sql/conn_executor_exec.go:333 +0x73f
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func2(0xc00ca5f5a8, 0xc00511e508, 0xc01861b6d8, 0xc01861b6c8, 0xc00ca5f6b0, 0xc00ca5f378, 0xc00ca5f728, 0xc01861b6f8)
	pkg/sql/conn_executor.go:2586 +0x1429
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc00511e508)
	pkg/sql/conn_executor.go:2588 +0x11dd
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00511e508, {0xd161690, 0xc00e97b2c0}, 0xc00c7e20b0, 0xc00e86cbb8, 0xc01484dcc0)
	pkg/sql/conn_executor.go:2387 +0x409
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc005443808, {0xd161690, 0xc00e97b2c0}, {0xc0171ec4d0?}, 0xc00e86cbb8, 0xc01484dcc0)
	pkg/sql/conn_executor.go:1128 +0x205
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommands(0xc0037fc408, {0xd161690, 0xc00e97b2c0}, {0x0, 0x4, {0x1, {0xb8f5f7a, 0x3}, {0xc00d9c2cc0, 0x18}, ...}, ...}, ...)
	pkg/sql/pgwire/conn.go:254 +0x728
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl.func4()
	pkg/sql/pgwire/server.go:1295 +0x211
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl in goroutine 8088
	pkg/sql/pgwire/server.go:1292 +0xa39
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	pkg/storage/disk_map.go:170
*   | github.com/cockroachdb/cockroach/pkg/storage.(*pebbleMap).Close
*   | 	pkg/storage/disk_map.go:186
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/producer.(*OrderedBuffer).Close
*   | 	pkg/crosscluster/producer/ordered_buffer.go:278
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/producer.(*OrderedStreamHandler).Close
*   | 	pkg/crosscluster/producer/ordered_event_stream.go:195
*   | github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*projectSetProcessor).close
*   | 	pkg/sql/rowexec/project_set.go:356
*   | github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*projectSetProcessor).ConsumerClosed
*   | 	pkg/sql/rowexec/project_set.go:365
*   | github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).InternalClose
*   | 	pkg/sql/execinfra/processorsbase.go:983
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Close
*   | 	pkg/sql/colexec/columnarizer.go:301
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec.(*invariantsChecker).Close
*   | 	pkg/sql/colexec/invariants_checker.go:137
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecargs.(*CloserRegistry).Close.func1
*   | 	pkg/sql/colexec/colexecargs/closer_registry.go:47
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
*   | 	pkg/sql/colexecerror/error.go:162
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecargs.(*CloserRegistry).Close
*   | 	pkg/sql/colexec/colexecargs/closer_registry.go:45
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).cleanup
*   | 	pkg/sql/colflow/vectorized_flow.go:674
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Cleanup
*   | 	pkg/sql/colflow/vectorized_flow.go:386
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run.func3
*   | 	pkg/sql/distsql_running.go:1022
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
*   | 	pkg/sql/distsql_running.go:1088
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
*   | 	pkg/sql/distsql_running.go:2397
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll.func3
*   | 	pkg/sql/distsql_running.go:2096
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll
*   | 	pkg/sql/distsql_running.go:2099
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine
*   | 	pkg/sql/conn_executor_exec.go:3694
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine
*   | 	pkg/sql/conn_executor_exec.go:3185
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
*   | 	pkg/sql/conn_executor_exec.go:1178
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func2
*   | 	pkg/sql/conn_executor_exec.go:183
* Wraps: (7) found 1 short-living non-stopped monitors in exec
*   | exec 20 KiB
*   |     flow-3064688e 20 KiB
* Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *safedetails.withSafeDetails (4) *withstack.withStack (5) *assert.withAssertionFailure (6) *withstack.withStack (7) *errutil.leafError
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnModeSmoketest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62808

pkg/crosscluster/logical/txnmode/txnmode_test.TestTxnModeSmoketest [failed](https://mesolite.cluster.engflow.com/invocations/default/4bb10cd2-a812-4676-8901-7d0cf82297f3?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY3Jvc3NjbHVzdGVyL2xvZ2ljYWwvdHhubW9kZTp0eG5tb2RlX3Rlc3Q=) on master @ [f9bcbc4fdf0ba5236f6312c00becf303c29f98f9](https://github.com/cockroachdb/cockroach/commits/f9bcbc4fdf0ba5236f6312c00becf303c29f98f9):

Fatal error:

```
panic: pebble: closed [recovered, repanicked]
	panic: found 1 short-living non-stopped monitors in exec
	exec 20 KiB
	    flow-c0bf04a6 20 KiB
	 [recovered]
	panic: found 1 short-living non-stopped monitors in exec
	exec 20 KiB
	    flow-c0bf04a6 20 KiB
	

```
Stack: 

```
goroutine 8717 [running]:
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).doStop(0xc005377760, {0xd16bc30, 0xc01380ec80}, 0x0)
	pkg/util/mon/bytes_usage.go:781 +0xa9c
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).EmergencyStop(...)
	pkg/util/mon/bytes_usage.go:717
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).close(0xc0118a2f08, {0xd16bc30, 0xc01380ec80}, 0x1)
	pkg/sql/conn_executor.go:1520 +0xdf8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc0118a2f08, {0xd16bc30, 0xc01380ec80}, {0xb1ef400, 0xc013a28d80})
	pkg/sql/conn_executor.go:1408 +0x227
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1({0xd16bc30, 0xc01380ec80}, {0x4b60a5?})
	pkg/sql/conn_executor.go:1126 +0x4b
panic({0xb1ef400?, 0xc013a28d80?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).doStop(0xc005377760, {0xd16bc30, 0xc00ebc5cc0}, 0x1)
	pkg/util/mon/bytes_usage.go:781 +0xa9c
github.com/cockroachdb/cockroach/pkg/util/mon.(*BytesMonitor).Stop(...)
	pkg/util/mon/bytes_usage.go:722
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine.func1()
	pkg/sql/conn_executor_exec.go:2932 +0x3c5
panic({0xb2e50a0?, 0xc012874f90?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError.func1()
	pkg/sql/colexecerror/error.go:137 +0x81a
panic({0xb2e50a0?, 0xc012874f90?})
	GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/pebble.(*DB).applyInternal(0xc004740008, 0xc0154aaa80, 0x134060fe, 0x0)
	external/com_github_cockroachdb_pebble/db.go:789 +0x9fd
github.com/cockroachdb/pebble.(*DB).Apply(...)
	external/com_github_cockroachdb_pebble/db.go:764
github.com/cockroachdb/pebble.(*DB).DeleteRange(0xc004740008, {0xc014ba65e8, 0x1, 0x18}, {0xc0104254b8, 0x1, 0x1}, 0x134060fe)
	external/com_github_cockroachdb_pebble/db.go:667 +0xc5
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleMap).Clear(0xc0025dcde0)
	pkg/storage/disk_map.go:170 +0x1b3
github.com/cockroachdb/cockroach/pkg/storage.(*pebbleMap).Close(0xc0025dcde0, {0xd16bc30, 0xc00eab6140})
	pkg/storage/disk_map.go:186 +0x45
github.com/cockroachdb/cockroach/pkg/crosscluster/producer.(*OrderedBuffer).Close(...)
	pkg/crosscluster/producer/ordered_buffer.go:278
github.com/cockroachdb/cockroach/pkg/crosscluster/producer.(*OrderedStreamHandler).Close(0xc0025411a0, {0xd16bc30, 0xc00eab6140})
	pkg/crosscluster/producer/ordered_event_stream.go:195 +0x10c
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*projectSetProcessor).close(0xc0115a4008)
	pkg/sql/rowexec/project_set.go:356 +0x13c
github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*projectSetProcessor).ConsumerClosed(0xc0115a4008)
	pkg/sql/rowexec/project_set.go:365 +0x27
github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).InternalClose(0xc011d59c08)
	pkg/sql/execinfra/processorsbase.go:983 +0x107
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Close(0xc011d59c08, {0xc0023c60af?, 0xc010425760?})
	pkg/sql/colexec/columnarizer.go:301 +0xb8
github.com/cockroachdb/cockroach/pkg/sql/colexec.(*invariantsChecker).Close(0xc0109d6d70, {0xd16aec0, 0xc0109d6c80})
	pkg/sql/colexec/invariants_checker.go:137 +0x75
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecargs.(*CloserRegistry).Close.func1()
	pkg/sql/colexec/colexecargs/closer_registry.go:47 +0xd6
github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError(0xc010425950)
	pkg/sql/colexecerror/error.go:162 +0x79
github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecargs.(*CloserRegistry).Close(0xc0024ce5d0, {0xd16aec0, 0xc0109d6c80})
	pkg/sql/colexec/colexecargs/closer_registry.go:45 +0x132
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).cleanup(0xc018b32160, {0xd16aec0, 0xc0109d6c80})
	pkg/sql/colflow/vectorized_flow.go:674 +0x58
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Cleanup(0xc0109d6c30, {0xd16aec0, 0xc0109d6c80})
	pkg/sql/colflow/vectorized_flow.go:386 +0x1f7
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run.func3()
	pkg/sql/distsql_running.go:1022 +0x59
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc00ff5c580, {0xd16aec0, 0xc0109d6c80}, 0xc00fb0f2b0, 0xc00fa1f1e0, 0xc001bd2400, 0xc0123ef688, 0xc0118a3d20, 0xc00c7f0698)
	pkg/sql/distsql_running.go:1088 +0x1c53
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc00ff5c580, {0xd16bc30, 0xc00ebc5cc0}, 0xc0118a3d20, 0xc00fb0f2b0, 0xc00fa1f1e0, {{0xd1b3370, 0xc01726de00}, 0x0}, 0xc0123ef688, ...)
	pkg/sql/distsql_running.go:2397 +0x3a5
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll.func3(0xc0118a3748, 0xc00ff5c580, {0xd16bc30, 0xc00ebc5cc0}, 0xc0118a3d20, 0xc00fb0f2b0, 0xc0123ef688)
	pkg/sql/distsql_running.go:2096 +0x19d
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll(0xc00ff5c580, {0xd16bc30, 0xc00ebc5cc0}, 0xc0118a3d20, 0xc00fb0f2b0, 0xc0118a3748, 0xc0123ef688, 0x0)
	pkg/sql/distsql_running.go:2099 +0x359
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc0118a2f08, {0xd16bc30, 0xc00ebc5cc0}, 0xc0118a3748, 0x3, {0x7f694153fe98, 0xc004608760}, 0x0, 0xc00d95d948, 0x0)
	pkg/sql/conn_executor_exec.go:3694 +0xba5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc0118a2f08, {0xd16bc30, 0xc00ebc5cc0}, 0xc0118a3748, {0x7f694153fe98, 0xc004608760})
	pkg/sql/conn_executor_exec.go:3185 +0x270e
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc0118a2f08, {0xd16aec0, 0xc0109d6780}, {{0xd19a520, 0xc00f416370}, {0x0, 0x0, 0x0}, {0xc015115073, 0x34}, ...}, ...)
	pkg/sql/conn_executor_exec.go:1178 +0x69b6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func2({0xd16aec0, 0xc0109d6780})
	pkg/sql/conn_executor_exec.go:183 +0x185
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithProfiling(0xc0118a2f08, {0xd16aec0, 0xc0109d6780}, {0xd19a520, 0xc00f416370}, 0xc00fc5a600, 0xc00c7f2b20)
	pkg/sql/conn_executor_exec.go:4766 +0x997
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc0118a2f08, {0xd16aec0, 0xc0109d6780}, {{0xd19a520, 0xc00f416370}, {0x0, 0x0, 0x0}, {0xc015115073, 0x34}, ...}, ...)
	pkg/sql/conn_executor_exec.go:182 +0xdbe
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execPortal(0xc0118a2f08, {0xd16aec0, 0xc0109d6780}, {{0xc0151150e6, 0x0}, 0xc00fc5a600, {0xc0023c6200, 0x2, 0x2}, {0xc002ab034e, ...}, ...}, ...)
	pkg/sql/conn_executor_exec.go:333 +0x73f
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd.func2(0xc00c7f35a8, 0xc0118a2f08, 0xc0104296d8, 0xc0104296c8, 0xc00c7f36b0, 0xc00c7f3378, 0xc00c7f3728, 0xc0104296f8)
	pkg/sql/conn_executor.go:2586 +0x1429
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc0118a2f08)
	pkg/sql/conn_executor.go:2588 +0x11dd
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc0118a2f08, {0xd16bc30, 0xc01380ec80}, 0xc00d59b600, 0xc014a04ff0, 0xc0164d2ad0)
	pkg/sql/conn_executor.go:2387 +0x409
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc00d518808, {0xd16bc30, 0xc01380ec80}, {0xc001fc5a20?}, 0xc014a04ff0, 0xc0164d2ad0)
	pkg/sql/conn_executor.go:1128 +0x205
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommands(0xc004608008, {0xd16bc30, 0xc01380ec80}, {0x0, 0x4, {0x1, {0xb8ff61a, 0x3}, {0xc017bed400, 0x18}, ...}, ...}, ...)
	pkg/sql/pgwire/conn.go:254 +0x728
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl.func4()
	pkg/sql/pgwire/server.go:1295 +0x211
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*Server).serveImpl in goroutine 8665
	pkg/sql/pgwire/server.go:1292 +0xa39
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	pkg/storage/disk_map.go:170
*   | github.com/cockroachdb/cockroach/pkg/storage.(*pebbleMap).Close
*   | 	pkg/storage/disk_map.go:186
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/producer.(*OrderedBuffer).Close
*   | 	pkg/crosscluster/producer/ordered_buffer.go:278
*   | github.com/cockroachdb/cockroach/pkg/crosscluster/producer.(*OrderedStreamHandler).Close
*   | 	pkg/crosscluster/producer/ordered_event_stream.go:195
*   | github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*projectSetProcessor).close
*   | 	pkg/sql/rowexec/project_set.go:356
*   | github.com/cockroachdb/cockroach/pkg/sql/rowexec.(*projectSetProcessor).ConsumerClosed
*   | 	pkg/sql/rowexec/project_set.go:365
*   | github.com/cockroachdb/cockroach/pkg/sql/execinfra.(*ProcessorBaseNoHelper).InternalClose
*   | 	pkg/sql/execinfra/processorsbase.go:983
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec.(*Columnarizer).Close
*   | 	pkg/sql/colexec/columnarizer.go:301
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec.(*invariantsChecker).Close
*   | 	pkg/sql/colexec/invariants_checker.go:137
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecargs.(*CloserRegistry).Close.func1
*   | 	pkg/sql/colexec/colexecargs/closer_registry.go:47
*   | github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError
*   | 	pkg/sql/colexecerror/error.go:162
*   | github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecargs.(*CloserRegistry).Close
*   | 	pkg/sql/colexec/colexecargs/closer_registry.go:45
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).cleanup
*   | 	pkg/sql/colflow/vectorized_flow.go:674
*   | github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlow).Cleanup
*   | 	pkg/sql/colflow/vectorized_flow.go:386
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run.func3
*   | 	pkg/sql/distsql_running.go:1022
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run
*   | 	pkg/sql/distsql_running.go:1088
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun
*   | 	pkg/sql/distsql_running.go:2397
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll.func3
*   | 	pkg/sql/distsql_running.go:2096
*   | github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRunAll
*   | 	pkg/sql/distsql_running.go:2099
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine
*   | 	pkg/sql/conn_executor_exec.go:3694
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine
*   | 	pkg/sql/conn_executor_exec.go:3185
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState
*   | 	pkg/sql/conn_executor_exec.go:1178
*   | github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt.func2
*   | 	pkg/sql/conn_executor_exec.go:183
* Wraps: (7) found 1 short-living non-stopped monitors in exec
*   | exec 20 KiB
*   |     flow-c0bf04a6 20 KiB
* Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *safedetails.withSafeDetails (4) *withstack.withStack (5) *assert.withAssertionFailure (6) *withstack.withStack (7) *errutil.leafError
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnModeSmoketest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/server/server_controller_sql.go
+++ b/pkg/server/server_controller_sql.go
@@ -15,6 +15,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/pgwire/pgwirecancel"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/catconstants"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/stop"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil/singleflight"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
@@ -182,5 +183,11 @@ func (t *tenantServerWrapper) serveConn(
 ) error {
 	pgCtx := t.server.sqlServer.AnnotateCtx(context.Background())
 	pgCtx = logtags.AddTags(pgCtx, logtags.FromContext(ctx))
+	stopCtx, stopHandle, err := t.stopper.GetHandle(pgCtx, stop.TaskOpts{TaskName: "serve-conn"})
+	if err != nil {
+		return errors.Wrap(err, "error getting stop handle")
+	}
+	handle := stopHandle.Activate(stopCtx)
+	defer handle.Release(stopCtx)
 	return t.server.sqlServer.pgServer.ServeConn(pgCtx, conn, status)
 }

